pkg/server/grpc/server.go

```diff
@@ -92,14 +117,21 @@ func (s *Server) Run() error {
 // New creates a new instance of `IstioCAServiceServer`.
 func New(ca ca.CertificateAuthority, hostname string, port int) *Server {
 	return &Server{
-		ca:       ca,
-		hostname: hostname,
-		port:     port,
+		authenticator: &clientCertAuthenticator{},
+		authorizer:    &simpleAuthorizer{},
+		ca:            ca,
+		hostname:      hostname,
+		port:          port,
 	}
 }
 
 func (s *Server) createTLSServerOption() grpc.ServerOption {
+	cp := x509.NewCertPool()
+	cp.AppendCertsFromPEM(s.ca.GetRootCertificate())
+
 	config := &tls.Config{
+		ClientCAs:  cp,
+		ClientAuth: tls.VerifyClientCertIfGiven,
```

Comment:
This logic needs to be carefully handled (i.e., for on prem, we will check the client cert must be presented, and for gcp, the jwt must be presented). Does not have to be in this PR, but just a reminder for both of us :)